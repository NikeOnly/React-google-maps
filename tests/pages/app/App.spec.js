import React from 'react'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import ConnectedApp, { App } from '../../../src/pages/app/App';

describe('App', () => {
    test('match snapshot', () => {
        const tree = renderer.create(<App loadPlacesList={jest.fn()} />).toJSON();

        expect(tree).toMatchSnapshot();
    })

    test('render connected component', () => {
        const initialState = { placesList: [] };
        const mockStore = configureStore();
        const store = mockStore(initialState);

        const container = shallow(<ConnectedApp store={store} /> );

        expect(toJson(container)).toMatchSnapshot();
        expect(container.prop('placesList')).toEqual(initialState.placesList);
    })
})
