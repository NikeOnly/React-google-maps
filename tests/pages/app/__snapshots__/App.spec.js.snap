// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div
  className="app-container"
>
  <ul />
  <div
    className="map-loading-element"
  />
</div>
`;

exports[`App render connected component 1`] = `
<App
  loadPlacesList={[Function]}
  placesList={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
